pkg/controller/apmserver/deployment_test.go

```diff
@@ -177,14 +182,24 @@ func expectedDeploymentParams() testParams {
 						},
 						Resources: DefaultResources,
 					}},
-					AutomountServiceAccountToken: &false,
+					AutomountServiceAccountToken: ptrFalse(),
 				},
 			},
 			Replicas: 0,
 		},
 	}
 }
 
+func withAssociations(as *apmv1.ApmServer, esAssocConf, kbAssocConf *commonv1.AssociationConf) *apmv1.ApmServer {
+	(&apmv1.ApmEsAssociation{
+		ApmServer: as,
+	}).SetAssociationConf(esAssocConf)
```

Comment:
feels like a constructor could improve readability a bit:
```
apmv1.NewApmEsAssociation(as).SetAssociationConf(esAssocConf)
```

but def not a big deal if we need it only in unit tests.